### ISSUE TITLE
Calling ClassInfo.getResource().open() after scanResult is closed results in NPE

### ISSUE THREAD

**Issue by sebthom** [classgraph/classgraph#885]
Calling `ClassInfo.getResource().open()` after the ScanResult is closed results in a ClassGraph internal NullPointerException.

Reproducer:
```java
package example;
import java.io.IOException;
import io.github.classgraph.*;

class ClassGraphTest {

   @org.junit.jupiter.api.Test
   void testClassGraph() throws IOException {
      final var className = ClassGraphTest.class.getName();
      final ClassInfo classInfo;
      try (ScanResult scanResult = new ClassGraph() //
         .enableAllInfo() //
         .acceptClasses(className) //
         .scan()) {
           classInfo = scanResult.getClassInfo(className);
      }

      try (var res = classInfo.getResource();
           var is = res.open()) { // <--- NPE thrown here
         // do interesting stuff
      }
   }
}
```

Exception:
```py
java.lang.NullPointerException: Cannot invoke "java.util.Set.add(Object)" because "this.openSlices" is null
	at nonapi.io.github.classgraph.fastzipfilereader.NestedJarHandler.markSliceAsOpen(NestedJarHandler.java:517)
	at nonapi.io.github.classgraph.fileslice.PathSlice.<init>(PathSlice.java:152)
	at io.github.classgraph.ClasspathElementDir$2.openAndCreateSlice(ClasspathElementDir.java:295)
	at io.github.classgraph.ClasspathElementDir$2.open(ClasspathElementDir.java:249)
	at example.ClassGraphTest.testClassGraph(ClassGraphTest.java:25)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

Interestingly the same works for JDK built-ins like "java.lang.Integer" and does not throw an NPE

---
Tested with OpenJDK 17 and classgraph 4.8.176

**Comment by lukehutch** [comment]
Thanks for the report. The docs are pretty clear, I think, that you can't use anything derived from a `ScanResult` after the `ScanResult` is closed, but I'll check if they can be improved. There's no easy way around this, because not closing resources after the `ScanResult` is closed would result in resources being leaked. At some point I would love to make it so that all resources are closed after a scan, and only opened on a per-file basis as resources are accessed, but that would be quite a lot of work, and I'm not actively working on ClassGraph anymore.

I committed a change to throw an `IllegalStateException` in this case rather than an NPE.

**Comment by lukehutch** [comment]
The change is released in 4.8.177. Since it still throws an exception in this case, it won't affect much from your point of view, but NPEs are always unacceptable!

**Comment by sebthom** [comment]
@lukehutch thanks for the fix! I also think an IllegalStateException is better than an NPE.
